src/plugins/ui_actions/public/service/ui_actions_service.ts

```diff
@@ -60,16 +60,16 @@ export class UiActionsService {
     return trigger.contract;
   };
 
-  public readonly registerAction = <A extends ActionDefinition>(
-    definition: A
-  ): Action<ActionContext<A>> => {
+  public readonly registerAction = <Context extends object>(
+    definition: ActionDefinition<Context>
+  ) => {
```

Comment:
this thing seem to be return `ActionInternal` now instead of `Action`. Seems incorrect

Reply:
That was the goal. `Action` type is the legacy type, which would nice to stop using. But also exposing `*Internal` is not the best, I guess I need to redefine `Action` to be a subset of `ActionInternal` interface.